storage/fanout.go

```diff
@@ -583,33 +599,22 @@ func (h *genericSeriesSetHeap) Pop() interface{} {
 // with "almost" the same data, e.g. from 2 Prometheus HA replicas. This is fine, since from the Prometheus perspective
 // this never happens.
 //
-// NOTE: Use this only when you see potentially overlapping series, as this introduces small overhead to handle overlaps
-// between series.
-func ChainedSeriesMerge(s ...Series) Series {
-	if len(s) == 0 {
+// NOTE: Use returned merge function only when you see potentially overlapping series, as this introduces small overhead
```

Comment:
I don't understand. This doesn't return a merge function. Perhaps this change in doc comment should have only happened for `NewCompactingChunkSeriesMerger` below?

Nits: "Use the returned..." and "...this introduces a small overhead...".

Reply:
You are right. It does not return function it is such function ;p Will fix!